### PR TITLE
Streamline version handling

### PR DESCRIPTION
**Pull request by pvizeli**
## Proposed change

dev upstream channel work like beta/stable now. It allows us to test the code and make sure it works. Now we can produce nightly with auto-update on dev channel and release automatic to beta.

After this PR, Supervisor follow CalVer.

## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (which adds functionality to the supervisor)
- [x] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Additional information

- This PR fixes or closes issue: fixes #
- This PR is related to issue:
- Link to documentation pull request: https://github.com/home-assistant/developers.home-assistant/pull/696

## Checklist

- [ ] The code change is tested and works locally.
- [ ] Local tests pass. **Your PR cannot be merged unless tests pass**
- [ ] There is no commented out code in this PR.
- [ ] I have followed the [development checklist][dev-checklist]
- [ ] The code has been formatted using Black (`black --fast supervisor tests`)
- [ ] Tests have been added to verify that the new code works.

If API endpoints of add-on configuration are added/changed:

- [ ] Documentation added/updated for [developers.home-assistant.io][docs-repository]

[dev-checklist]: https://developers.home-assistant.io/docs/en/development_checklist.html
[docs-repository]: https://github.com/home-assistant/developers.home-assistant

### CODE REVIEW

File: supervisor/core.py [hunk 1]
```diff
@@ -97,15 +97,6 @@ async def connect(self):
         # Check supervisor version/update
         if self.sys_dev:
             self.sys_config.version = self.sys_supervisor.version
-        elif (
```
Comment (ludeeus) [inline supervisor/core.py:100]: Why was this removed?

File: supervisor/addons/addon.py [hunk 2]
```diff
@@ -149,6 +149,11 @@ def version(self) -> Optional[str]:
         """Return installed version."""
         return self.persist[ATTR_VERSION]
 
+    @property
+    def need_update(self) -> bool:
+        """Return True if an update is available."""
+        return self.version != self.latest_version
```
Comment (ludeeus) [inline supervisor/addons/addon.py:155]: Looking at the UI implementation, we don't show an update if add-on is running detached, should we guard against that her instead?

Comment (pvizeli) [reply supervisor/addons/addon.py:155]: right

File: supervisor/core.py (hunk 1 above)
Comment (pvizeli) [reply supervisor/core.py:100]: Is not needed anymore with the new build logic

File: supervisor/api/addons.py [hunk 3]
```diff
@@ -161,8 +162,12 @@ async def list(self, request: web.Request) -> Dict[str, Any]:
                 ATTR_DESCRIPTON: addon.description,
                 ATTR_ADVANCED: addon.advanced,
                 ATTR_STAGE: addon.stage,
-                ATTR_VERSION: addon.latest_version,
-                ATTR_INSTALLED: addon.version if addon.is_installed else None,
+                ATTR_VERSION: addon.version if addon.is_installed else None,
+                ATTR_VERSION_LATEST: addon.latest_version,
+                ATTR_UPDATE_AVAILABLE: addon.need_update
+                if addon.is_installed
+                else None,
```
Comment (ludeeus) [inline supervisor/api/addons.py:169]:
```suggestion
                else False,
```
When it's not installed, there are no updates, if this is None, we need to check if installed in the UI before we check this

File: supervisor/api/addons.py [hunk 4]
```diff
@@ -209,6 +214,7 @@ async def info(self, request: web.Request) -> Dict[str, Any]:
             ATTR_REPOSITORY: addon.repository,
             ATTR_VERSION: None,
             ATTR_VERSION_LATEST: addon.latest_version,
+            ATTR_UPDATE_AVAILABLE: None,
```
Comment (ludeeus) [inline supervisor/api/addons.py:217]:
```suggestion
            ATTR_UPDATE_AVAILABLE: False,
```

**Review by ludeeus — APPROVED** [review]

File: supervisor/supervisor.py [hunk 5]
```diff
@@ -50,7 +51,10 @@ def ip_address(self) -> IPv4Address:
     @property
     def need_update(self) -> bool:
         """Return True if an update is available."""
-        return self.version != self.latest_version
+        try:
```
Comment (ludeeus) [inline supervisor/supervisor.py:54]:
```suggestion
		if self.version == "DEV":
			return False
        try:
```

Let's add a check for DEV as the version here to stop it from offering updates when you start from git

Comment (pvizeli) [reply supervisor/supervisor.py:54]: I like that. Let me write it self on vscode

Comment (ludeeus) [reply supervisor/supervisor.py:54]: If you do that, maybe add it as a const

**Review by ludeeus — APPROVED** [review]
